frontend/public/components/modals/configure-machine-autoscaler-modal.tsx

```diff
@@ -76,65 +97,69 @@ const ConfigureMachineAutoscalerModal = (props: ConfigureMachineAutoscalerModalP
         })
         .catch(() => {});
     },
-    [props, createAutoscaler, handlePromise, navigate],
+    [createAutoscaler, handlePromise, navigate, closeOverlay, close],
   );
 
   const {
     machineSet: {
       metadata: { name },
     },
-    cancel,
   } = props;
   const { t } = useTranslation();
 
   return (
-    <form onSubmit={submit} name="form" className="modal-content">
-      <ModalTitle className="modal-header">{t('public~Create MachineAutoscaler')}</ModalTitle>
+    <Modal isOpen onClose={closeOverlay} variant="small">
+      <ModalHeader
+        title={t('public~Create MachineAutoscaler')}
+        labelId="configure-machine-autoscaler-modal-title"
+        description={t('public~This will automatically scale machine set {{ name }}.', { name })}
+      />
       <ModalBody>
-        <p>
-          <Trans t={t} ns="public">
-            This will automatically scale machine set <b>{{ name }}</b>.
-          </Trans>
-        </p>
-        <div className="form-group">
-          <label>
-            {t('public~Minimum replicas:')}
-            <NumberSpinner
-              value={minReplicas}
-              onChange={changeMinReplicas}
-              changeValueBy={changeMinReplicasBy}
-              autoFocus
-              required
-            />
-          </label>
-        </div>
-        <div className="form-group">
-          <label>
-            {t('public~Maximum replicas:')}
-            <NumberSpinner
-              value={maxReplicas}
-              onChange={changeMaxReplicas}
-              changeValueBy={changeMaxReplicasBy}
-              required
-            />
-          </label>
-        </div>
+        <Form>
+          <FormGroup>
+            <label>
```

Comment:
We could use the `label` prop of `FormGroup` instead of `<label>`   here.
```suggestion
      <FormGroup label={t('public~Minimum replicas:')} isRequired fieldId={`${id}-address`}>
```
Wondering about the `isRequired` prop here since we provide one in the `NumberSpinner` as well, we might need to adjust and remove the required in the `NumberSpinner` and add isRequired to the `FormGroup` - to keep the consistency.

Reply:
@Mylanos checked the codebase, the isRequired prop on FormGroup and the required prop on NumberSpinner serve different purposes, and you generally want to keep both. 
FormGroup's isRequired prop is a visual indicator - adds an asterisk to the field label. Also provides semantic information for screen readers. Least but not last it follows PatternFly design patterns.
NumberSpinner's required prop is a HTML5 validation and gets passed to the underlying <input> element as the required attribute. Enables native browser form validation and prevents form submission if the field is empty